### PR TITLE
remove modules that is the relative path.

### PR DESCRIPTION
**Pull request by jquense**
What is this for? It breaks a lot stuff in weird subtle ways. e.g. react-popper requires a file `popper.js` from node_modules in `Popper.js`, this setting causes the import to resolve to itself instead of the other module of a similar name.

### CODE REVIEW

**Comment by pieh** [comment]
some (not a lot) context for this - https://github.com/gatsbyjs/gatsby/pull/7480 - I would be very happy to see this gone, but css modules stuff would probably need to be looked at again

**Comment by pieh** [comment]
maybe removing entire `modules` entry would actually fix that instead of this ugly hack I added ...

**Comment by jquense** [comment]
> but css modules stuff would probably need to be looked at again 

what is the check for that, i use css modules in all my gatsby projects, including ones where i remove this manually and haven't seen a problem. maybe i'm not hitting the edge case?

**Comment by pieh** [comment]
I've used this very basic reproduction https://github.com/timurc/gatsby-starter-default-css-module-image-test/ which was throwing can't find 'boa.png' ( https://github.com/timurc/gatsby-starter-default-css-module-image-test/blob/master/src/components/style.module.css - and image is sibling file to css module file)

**Comment by pieh** [comment]
what was also weird about this issue - that only combo of css modules + mini extract plugin was resulting in that - development (so style-loader instead of mini extract plugin) didn't have problem

**Comment by jquense** [comment]
BAH found the problem, is cssnano, running to early to minimize the styles and "cleaning" the url to a non relative one. fix incoming

**Comment by jquense** [comment]
OK fixed and minification is now done correctly per the webpack docs

File: packages/gatsby/src/utils/webpack-utils.js [hunk 1]
```diff
@@ -1,10 +1,10 @@
 // @flow
-const os = require(`os`)
 
 const autoprefixer = require(`autoprefixer`)
 const flexbugs = require(`postcss-flexbugs-fixes`)
-const UglifyPlugin = require(`uglifyjs-webpack-plugin`)
+const TerserPlugin = require(`terser-webpack-plugin`)
```
Comment (jquense) [inline packages/gatsby/src/utils/webpack-utils.js]: this is a drop-in replacement but maintained unlike uglify-es

File: packages/gatsby/src/utils/webpack.config.js [hunk 2]
```diff
@@ -430,7 +415,7 @@ module.exports = async (
     // https://github.com/defunctzombie/package-browser-field-spec); setting
     // the target tells webpack which file to include, ie. browser vs main.
     target: stage === `build-html` || stage === `develop-html` ? `node` : `web`,
-    profile: stage === `production`,
```
Comment (jquense) [inline packages/gatsby/src/utils/webpack.config.js]: i don't think this was ever `true`? stage can't be `production`

Comment (KyleAMathews) [reply packages/gatsby/src/utils/webpack.config.js]: haha yeah 🤔

File: packages/gatsby/package.json [hunk 3]
```diff
@@ -112,6 +112,7 @@
     "socket.io": "^2.0.3",
     "string-similarity": "^1.2.0",
     "style-loader": "^0.21.0",
+    "terser-webpack-plugin": "^1.0.2",
     "type-of": "^2.0.1",
     "uglifyjs-webpack-plugin": "^1.2.4",
```
Comment (pieh) [inline packages/gatsby/package.json]: this can be removed?

File: packages/gatsby/src/utils/webpack-utils.js [hunk 4]
```diff
@@ -428,23 +421,25 @@ module.exports = async ({
    * Minify javascript code without regard for IE8. Attempts
    * to parallelize the work to save time. Generally only add in Production
    */
-  plugins.uglify = ({ uglifyOptions, ...options } = {}) =>
-    new UglifyPlugin({
+  plugins.minifyJs = ({ terserOptions, ...options } = {}) =>
+    new TerserPlugin({
       cache: true,
-      parallel: os.cpus().length - 1,
+      parallel: true,
```
[the remark this answers is not in the document]
Comment (jquense) [reply packages/gatsby/src/utils/webpack-utils.js]: you can specify a number here, but `true` defaults to `os.cpus().length - 1` so no need to specify it ourselves

File: packages/gatsby/src/utils/webpack.config.js [hunk 5]
```diff
@@ -455,7 +440,11 @@ module.exports = async (
       splitChunks: {
         name: false,
       },
-      minimize: !program.noUglify,
```
[the remark this answers is not in the document]
Comment (jquense) [reply packages/gatsby/src/utils/webpack.config.js]: I moved it lower. `noUglify` here would disable css minimization as well, which isn't what the argument suggests would happen

File: packages/gatsby/src/utils/webpack-utils.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (DSchau) [reply packages/gatsby/src/utils/webpack-utils.js]: Sweet. Good catch!

**Comment by jquense** [comment]
🎉 THANKS YA'LL
